src/test/groovy/graphql/ExecutionInputTest.groovy

```diff
@@ -111,6 +135,31 @@ class ExecutionInputTest extends Specification {
         executionInput.query == "new query"
     }
 
+    def "transform works and copies values with raw variables"() {
+        when:
+        def executionInputOld = ExecutionInput.newExecutionInput().query(query)
+                .dataLoaderRegistry(registry)
+                .cacheControl(cacheControl)
+                .rawVariables(rawVariables)
+                .extensions([some: "map"])
+                .root(root)
+                .graphQLContext({ it.of(["a": "b"]) })
+                .locale(Locale.GERMAN)
+                .build()
+        def graphQLContext = executionInputOld.getGraphQLContext()
+        def executionInput = executionInputOld.transform({ bldg -> bldg.query("new query") })
+
+        then:
+        executionInput.graphQLContext == graphQLContext
+        executionInput.root == root
+        executionInput.rawVariables == rawVariables
+        executionInput.dataLoaderRegistry == registry
+        executionInput.cacheControl == cacheControl
+        executionInput.locale == Locale.GERMAN
+        executionInput.extensions == [some: "map"]
+        executionInput.query == "new query"
+    }
```

Comment:
What happens if we set both

```
.rawVariables(x).variables(y)
```

Should we have a test for this.  Do they go into the same place?

Reply:
I'll add a test.

Yes, in `ExecutionInput` both `rawVariables` and `variables` set the same field. If both get set, the last one would be used (in your example `y`).

Is it worthwhile preventing both `.rawVariables(x).variables(y)` to be used? I don't mind adding another check if it'll be clearer